web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java

```diff
@@ -220,4 +223,16 @@ public void setCookieMaxAge(int cookieMaxAge) {
 		this.cookieMaxAge = cookieMaxAge;
 	}
 
+	/**
+	 * Add the "SameSite" attribute to the generated cookie.
+	 * <p>This limits the scope of the cookie such that it will only be
+	 * attached to same site requests if {@code "Strict"} or cross-site
+	 * requests if {@code "Lax"}.
+	 * @since 6.1
+	 * @see <a href="https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis#section-4.1.2.7">RFC6265 bis</a>
+	 */
+	public void setSameSite(String sameSite) {
```

Comment:
I wonder if instead of adding a new method each time a piece of the cookie needs altering, a new method like this should be added:

```java
public void setResponseCookieCustomizer(Consumer<ResponseCookieBuilder> responseCookieCustomizer)
```

Then, this code could instead do:

```java
this.responseCookieCustomizer.accept(builder);
ResponseCookie rCookie = builder.build();
```

Then, the other setter methods, e.g. `setSecure`, `setCookieMaxAge`, etc. could be deprecated.

Reply:
I implemented it using almost the same code that you provided as a demo. I agree that the final result is way better than adding new getters.